sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -422,3 +433,49 @@ def _create_column_comments(
                 f"Column comments for table '{table.alias_or_name}' not registered - this may be due to limited permissions.",
                 exc_info=True,
             )
+
+    def get_alter_expressions(
+        self, current_table_name: TableName, target_table_name: TableName
+    ) -> t.List[exp.Alter]:
+        schema_expressions = super().get_alter_expressions(current_table_name, target_table_name)
+        additional_expressions = []
+
+        # check for a change in clustering
+        current_table = exp.to_table(current_table_name)
+        target_table = exp.to_table(target_table_name)
+
+        current_table_info = t.cast(
+            SnowflakeDataObject,
+            seq_get(self.get_data_objects(current_table.db, {current_table.name}), 0),
+        )
+        target_table_info = t.cast(
+            SnowflakeDataObject,
+            seq_get(self.get_data_objects(target_table.db, {target_table.name}), 0),
+        )
+
+        if current_table_info and target_table_info:
+            if any([current_table_info.is_clustered, target_table_info.is_clustered]):
+                if target_table_info.clustering_key:
+                    if current_table_info.clustering_key != target_table_info.clustering_key:
+                        # Note: If you create a table with eg `CLUSTER BY (c2, c1)` and read the info back from information_schema,
+                        # it gets returned as a string like "LINEAR(c2, c1)" which we need to parse back into a list of columns
+                        parsed_cluster_key = parse_one(
+                            target_table_info.clustering_key, dialect=self.dialect
+                        )
+                        additional_expressions.append(
+                            exp.Alter(
+                                this=current_table,
+                                kind="TABLE",
+                                actions=[exp.Cluster(expressions=parsed_cluster_key.expressions)],
+                            )
+                        )
+                else:
```

Comment:
I feel like we can do `elif current_table_info.is_clustered:` here and get rid of the `any` block

Reply:
Fair call, i've updated it